### PR TITLE
Fixes #31719 - Fix array settings to accept empty array

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const SettingForm = ({
     let submitValues = { setting: values };
 
     if (setting && setting.settingsType === 'array') {
-      const splitValue = { value: values.value.split(',') };
+      const splitValue = { value: values.value === '' ? [] : values.value.split(',') };
```
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js]: In most languages `''.split(',')` returns `[]` but looks like Javascript is weird.

**Review by ekohl — DISMISSED** [review]

**Review by ekohl — DISMISSED** [review]
Looks like the last change was just a formatting change.

File: webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js [hunk 2]
```diff
@@ -18,7 +18,9 @@ const SettingForm = ({
     let submitValues = { setting: values };
 
     if (setting && setting.settingsType === 'array') {
-      const splitValue = { value: values.value.split(',') };
+      const splitValue = {
+        value: values.value === '' ? [] : values.value.split(','),
+      };
       submitValues = { setting: splitValue };
```
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js]: Btw, is this the same?
```suggestion
      const splitValue = values.value === '' ? [] : values.value.split(',');
      submitValues = { setting: { value: splitValue } };
```

Comment (xprazak2) [reply webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js]: Yes, updated.

**Review by ekohl — APPROVED** [review]

**Comment by ezr-ondrej** [comment]
[test upgrade]

**Comment by ekohl** [comment]
I don't think upgrade could really be affected by this so I'm merging this.

**Comment by ekohl** [comment]
@xprazak2 this probably needs to be cherry picked into stable branches. Do you know which ones?

**Comment by ezr-ondrej** [comment]
CP 2.4: #8306

**Comment by tbrisker** [comment]
2.4 - 1bdefd70b

**Pull request merged**